### ISSUE TITLE
SSH timeout error on `suspend`

### ISSUE THREAD

**Issue by Saviq** [canonical/multipass#609]
Suspending an instance took longer than usual (`iowait` likely) and multipass threw an error out:
```
$ multipass suspend snapcraft-multipass                 
suspend failed: [ssh proc] failed to open session channel: '' 
```

**Comment by townsend2010** [comment]
This is really strange...I see nowhere in the code that we are trying to execute a process inside the instance while suspending which is the only place that `[ssh proc] failed to open session channel: ''` comes from.  I'm really scratching my head on this one.

That said, unrelated to the error given in the bug, we do have a hard timeout of 30 seconds when suspending, both in the daemon code and also snapd will `SIGKILL` `multipassd` if it doesn't die in 30 seconds.  I think I'm going to remove the daemon 30 second timeout since it really doesn't make sense to have that in there.  But that won't fix the error seen here...

**Comment by townsend2010** [comment]
Actually, what I think is happening is that the current suspend code runs another QEventLoop for the aforementioned timer and I'm guessing that the event loop processed something else that was going on, like a `list` or something from a running mount or something.  Nonetheless, I'm removing that timer and eventloop and just be done with it.

**Comment by townsend2010** [comment]
@Saviq, question...do you prefer the client wait on an instance to finish suspending, ie, with the spinner, or do you think that once the client issues suspend, it just returns and then state can be figured out by `multipass list` or `multipass info <instance_name>`?  This kind of goes along with the notion that `stop` should return immediately too (which it currently still doesn't).

**Comment by Saviq** [comment]
@townsend2010 all of our commands today are synchronous. If we want to change that, we should probably do it wholesale while introducing `--wait`/`--no-wait` options as applicable. So in this, I'd say we should wait.

**Comment by townsend2010** [comment]
@Saviq, ok, makes sense.  Thanks!

**Comment by sinke237** [comment]
I have the same issue now(May 24, 2024).
Strange behaviour of multipass.
When I 
`multipass exec more-human zsh`
It takes a very usually long time to execute.
If it succeeds to do so, I can not run any command inside the VM.
If it does not succeed, I get the response,
`exec failed: [ssh client] channel creation failed: ''`

any other time when i run any command using multipass, i get the error,
`exec failed: [ssh client] channel creation failed: ''`
